.github/workflows/cd.yml

```diff
@@ -0,0 +1,94 @@
+name: CD
+concurrency: cd
+
+# Trigger workflow on completed CI (further checks below)
+on:
+  workflow_run:
+    workflows: [CI]
+    types: [completed]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    # Skip unless CI was successful and ran on a ref starting with 'v' (release tag)
+    if: ${{ github.event.workflow_run.conclusion == 'success' && startsWith(github.event.workflow_run.head_branch, 'v') }}
+    # NOTE: This works because we currently only trigger CI on a push to the 'develop'
+    # branch or a 'v*'-tag, but it seems rather brittle.
+    # Unfortunately, there is not much more info we get from the CI workflow
+    # ('workflow_run') than the ref name. No ref, ref_type, etc., so we don't even know
+    # if a tag or a branch was pushed. :(
+    # See https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#workflow_run
+    # NOTE: (2) An alternative solution might be to restructure workflows, so that all
+    # test logic from 'ci.yml' is moved to a separate workflow file '_test.yml', that
+    # can be included in both CI (triggered on push to 'develop'-branch) and CD
+    # (triggered on push to 'v*'-tag) workflows.
+    outputs:
+      release_id: ${{ steps.gh-release.outputs.id }}
+    steps:
+      - name: Checkout release tag
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846
+        with:
+          ref: ${{ github.event.workflow_run.head_branch }}
+
+      - name: Set up Python
+        uses: actions/setup-python@0ebf233433c08fb9061af664d501c3f3ff0e9e20
+        with:
+          python-version: '3.x'
+
+      - name: Install build dependency
+        run: python3 -m pip install --upgrade pip build
+
+      - name: Build binary wheel and source tarball
+        run: python3 -m build --sdist --wheel --outdir dist/ .
+
+      - id: gh-release
+        name: Publish GitHub release candiate
+        uses: softprops/action-gh-release@1e07f4398721186383de40550babbdf2b84acfc5
+        with:
+          name: ${{ github.event.workflow_run.head_branch }}-rc
```

Comment:
so the release made here will have a "title" `vA.B.C-rc` but the download URLs will be the final ones already? This seems ok to me.

Does the release contain the annoying GitHub created "source code" zips as well or do they not get created with `action-gh-release`? (this is not important just wondering)

Reply:
> so the release made here will have a "title" `vA.B.C-rc` but the download URLs will be the final ones already? This seems ok to me.

Correct. The -rc suffix is only in the title. The URL uses the release tag, which is whatever's pushed.

> 
> Does the release contain the annoying GitHub created "source code" zips as well or do they not get created with `action-gh-release`? (this is not important just wondering)

Yes it does. I haven't checked if there's a way to prevent GitHub from doing this.  Will add a quick note in the list of TODO's above, which I intend to populate tickets from.